src/plugins/elasticsearch/lib/create_proxy.js

```diff
@@ -24,3 +22,10 @@ module.exports = function createProxy(server, method, route, config) {
   server.route(options);
 };
 
+createProxy.createPath = function createPath(path) {
+  const pre = '/elasticsearch';
+  const sep = path[0] === '/' ? '' : '/';
+  return `${pre}${sep}${path}`;
```

Comment:
I was mostly preserving the logic from the previous code, but you're definitely right.  I'll throw that into a follow up PR.

Note: that change actually went in via the PR that this depended on rather than this PR itself.